src/main/java/eu/europeana/metis/sandbox/domain/Record.java

```diff
@@ -101,25 +100,25 @@ public Language getLanguage() {
   /**
    * Content of the record
    *
-   * @implNote Overwriting this field contents after construction could cause problems.
-   * <br />
-   * We are not making a copy of it because it
-   * is expensive and Record object is expected to be use as a non mutable object
+   * @implNote Overwriting this field contents after construction could cause problems. <br /> We
+   * are not making a copy of it because it is expensive and Record object is expected to be use as
+   * a non mutable object
    */
   public byte[] getContent() {
-    return this.content;
+    return this.content; //NOSONAR
```

Comment:
Sonar complains because

> Mutable members should not be stored or returned directly

I am aware of this, but returning a copy of the byte array will be expensive.
But I started thinking if we include methods in metis libraries to handle byte array directly then same bug is going to start popping everywhere, I am starting to guess the reason in metis libraries to treat the record as a String is because of the immutability of it which makes it easier to work in trade for transform to and from byte array (which most metis libraries do). 
I started switching to byte array mostly because when I publish/consume messages from rabbit, the record xml representation travels as a byte[] but I treat it in the application as a String, this is actually common, because when you send content through Rabbit, Spring AMQP will set it as a byte[] and if you do not specify a custom MessageConverter then the transformation from and to a String will happen under the hoods, same happens with Kafka. 
So I am thinking between going back to Strings to take advantage of its immutability and pay the price of transformations between publish/consume in Rabbit (which is common anyways) and keep using metis libraries as they are now (which most of them take String as the record parameter) because transform back and forth String and byte[] is a concern Spring AMQP. 
Or we can push to change it all to byte[] but we need to understand that the content of a record could be altered and also I am not sure if metis libraries could have an impact on that change.
Anyways there is a story to address the change, I just started thinking if it is worth to do it, and what impact it could have.

Reply:
I understand. But perhaps we could choose a middle path. Instead of byte arrays, we could pass InputStreams to the library as well. If we change this method to return a bytearrayinputstream we get rid of the warning, and we can change the libraries to accept this instead of (or in addition to) byte arrays.
What do you think?
In the end, I think in this case performance considerations should trump code quality issues.

Reply:
I think `ByteArrayInputStream` sounds good, then metis libraries could take it instead of String or byte[]. And thinking again we could reconsider returning a copy of the byte[] (when byte[] is needed), at the end it will still be better than doing `String.getBytes()` 

Also I think while it will be faster we are actually not sure of the performance gains, we do not have measures of what we are winning with these changes (or at least I am not sure, maybe you already notice this in the past with ecloud team and have info on the improvements). So I will add to [MET-2680](https://europeana.atlassian.net/browse/MET-2680) a comment on running tests to measure the performance wins with these changes.
At the end I think we consider this because we will win something, but I want to understand what is that something we are winning